### DISCUSSION TITLE
Fulfillment withdrawal requests mapping

### DISCUSSION THREAD

**Discussion by jferrant** [stacks-network/stacks-core#3843] [sBTC]
How do we map a withdrawal request made on STX to a resulting BTC withdrawal fulfillment transaction?

This is not only useful for a user who wants to see their transaction progress, but also from the perspective of a signer needing to see whether a withdrawal has been successfully processed.

@netrome @soju-drinker @stjepangolemac @jcnelson @jo-tm

**Comment by netrome** [comment]
I guess the simple approach would be to map any fulfillment transactions on Bitcoin to the earliest unfulfilled withdrawal request. What problems exist with this approach?

Example1:
Alice requests to withdraw 5 BTC (tx1)
Alice requests to withdraw 10 BTC (tx2)
Two blocks later, there's a fulfillment to Alices address for 5 BTC, we map this one to tx1
Eight blocks later, there's a fulfillment to Alicess address for 10 BTC, we map this one to tx2

Example 2:
Alice requests to withdraw 5 BTC (tx1)
Alice requests to withdraw 10 BTC (tx2)
Two blocks later, there's a fulfillment to Alices address for 2 BTC, we map this one to tx1 as a partial fulfillment
Three blocks later, there's a fulfillment to Alices address for 3 BTC, we map this one to tx1 as a partial fulfillment
Eight blocks later, there's a fulfillment to Alicess address for 10  BTC, we map this one to tx2

Example 3:
Alice requests to withdraw 5 BTC (tx1)
Alice requests to withdraw 10 BTC (tx2)
Two blocks later, there's a fulfillment to Alices address for 15 BTC, we map this one to tx1 and tx2 as a multi-fulfillment.

The only other option I see is to add more data to the fulfillment transaction to point to the withdrawal request txid.

**Reply by stjepangolemac** [reply]
Is there a particular reason to support partial and bundled fulfillments or it's just nice to have?

**Reply by netrome** [reply]
For the time being it's a nice to have. It was discussed long ago as something we'd need for the 1.1 release, becuase for example recovery mode cannot guarantee 1:1 fulfillments.
